File: api/src/models/transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    travellerAddress: {
        type: String,
        required: true,
        trim: true
    },
    helperAddress: {
        type: String,
        required: true,
        trim: true
    },
    index: {
        type: Number
    },
    approval: {
        type: String,
        enum: ['Pending', 'Rejected', 'Approved'],
        default: 'Pending'
    },
    imageId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'Photo'
    }
});

const Transaction = mongoose.model('Transactions', transactionSchema);

module.exports = Transaction;

File: mobile_apps/lib/models/transaction.dart
class Transaction {
  int index;
  String travellerAddress;
  String helperAddress;
  String approval;
  String imageId;

  Transaction({
    this.index,
    this.travellerAddress,
    this.helperAddress,
    this.approval,
    this.imageId,
  });

  Transaction.fromJson(Map<String, dynamic> json)
      : index = json['index'],
        travellerAddress = json['travellerAddress'],
        helperAddress = json['helperAddress'],
        approval = json['approval'],
        imageId = json['imageId'];

  Map<String, dynamic> toJson() => {
        'index': index,
        'travellerAddress': travellerAddress,
        'helperAddress': helperAddress,
        'approval': approval,
        'imageId': imageId,
      };
}
